cpu/kinetis_common/include/mcg.h

```diff
@@ -0,0 +1,139 @@
+ *                  #define KINETIS_MCG_ERC_OSCILLATOR        1
+ *                  #define KINETIS_MCG_ERC_FRDIV             0
+ *                  #define KINETIS_MCG_ERC_RANGE             0
+ *                  #define KINETIS_MCG_ERC_FREQ              (32768U)
+ *
+ *
+ *              Example for FEE Mode, external 4MHz reference
+ *              (\f$MCGOUTCLK
+ *              = \frac{ERC_{\mathrm{freq}}\cdot FFL_{\mathrm{Factor}}}{FRDIV}
+ *              = \frac{4\,\mathrm{MHz}\cdot 732}{128} = 22.875\,\mathrm{MHz}\f$):
+ *
+ *                  #define KINETIS_MCG_USE_ERC               1
+ *                  #define KINETIS_MCG_USE_PLL               0
+ *                  #define KINETIS_MCG_DCO_RANGE             (24000000U)
+ *                  #define KINETIS_MCG_ERC_OSCILLATOR        0
+ *                  #define KINETIS_MCG_ERC_FRDIV             2
+ *                  #define KINETIS_MCG_ERC_RANGE             1
+ *                  #define KINETIS_MCG_ERC_FREQ              (4000000U)
+ *
+ *              Example for PEE Mode, external 4MHz reference
+ *              (\f$MCGOUTCLK
+ *              = \frac{ERC_{\mathrm{freq}}\cdot VDIV0}{PRDIV}
+ *              = \frac{4\,\mathrm{MHz}\cdot 24}{2} = 48\,\mathrm{MHz}\f$):
+ *
+ *                  #define KINETIS_MCG_USE_ERC               1
+ *                  #define KINETIS_MCG_USE_PLL               1
+ *                  #define KINETIS_MCG_DCO_RANGE             (24000000U)
+ *                  #define KINETIS_MCG_ERC_OSCILLATOR        0
+ *                  #define KINETIS_MCG_ERC_FRDIV             2
+ *                  #define KINETIS_MCG_ERC_RANGE             1
+ *                  #define KINETIS_MCG_ERC_FREQ              (4000000U)
+ *                  #define KINETIS_MCG_PLL_PRDIV             1
+ *                  #define KINETIS_MCG_PLL_VDIV0             0
+ *                  #define KINETIS_MCG_PLL_FREQ              (48000000U)
+ *
+ *
+ *              The desired mode can be selected in cpu.c:cpu_clock_init()
+ *              with kinetis_mcg_set_mode(KINETIS_MCG_PEE);
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the Kinetis MCG driver.
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ */
+
+#ifndef __MCG_CPU_H
+#define __MCG_CPU_H
+
+#include "cpu-conf.h"
+#include "periph_conf.h"
+
+#if KINETIS_CPU_USE_MCG
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+typedef enum kinetis_mcg_mode {
+    KINETIS_MCG_PEE = 0, /**< PLL Engaged External Mode */
+    KINETIS_MCG_PBE,     /**< PLL Bypassed External Mode */
+    KINETIS_MCG_BLPE,    /**< FLL Bypassed Low Power External Mode */
+    KINETIS_MCG_BLPI,    /**< FLL Bypassed Low Power Internal Mode */
+    KINETIS_MCG_FBE,     /**< FLL Bypassed External Mode */
+    KINETIS_MCG_FBI,     /**< FLL Bypassed Internal Mode */
+    KINETIS_MCG_FEE,     /**< FLL Engaged External Mode */
+    KINETIS_MCG_FEI,     /**< FLL Engaged Internal Mode */
+} kinetis_mcg_mode_t;
+
+uint32_t cpufreq;
```

Comment:
cpufreq is a zombie, it was planned that the mcg driver calculates current cpu frequency, but because of lack of time it has not been done. I will remove it for now.